lib/ui/painting.dart

```diff
@@ -1215,6 +1289,24 @@ class Image extends NativeFieldWrapperClass2 {
   /// The number of image pixels along the image's vertical axis.
   int get height native 'Image_height';
 
+  /// Converts the [Image] object into a byte array.
+  ///
+  /// The [format] is encoding format to be used.
+  ///
+  /// Returns a future which complete with the binary image data (e.g a PNG or JPEG binary data) or
+  /// an error if encoding fails.
+  Future<ByteData> toByteData({EncodingFormat format: const EncodingFormat.jpeg()}) {
+    return _futurize((_Callback<ByteData> callback) {
+      _toByteData(format._format, format._quality, (Uint8List encoded) {
```

Comment:
we're discarding the string returned by toByteData here, which seems bad

Reply:
Good catch.  We just need to `return _toByteData(...` here

Reply:
Done.